bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/RetryListener.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an implementation of {@link Response.CompleteListener}, which retries an HTTP Request, if it has failed.
+ *
+ * @author Svilen Valkanov - initial contribution
+ */
+
+public class RetryListener implements Response.CompleteListener {
```

Comment:
This class imho does not need to be public (and be in an exported package).
As discussed on the HttpUtil, I would tend to say we can remove it completely.